iceoryx_posh/CMakeLists.txt

```diff
@@ -219,16 +246,44 @@ if(TOML_CONFIG)
     endif(NOT cpptoml_FOUND)
     message("-- Build with TOML config file support.")
 
-    add_library(iceoryx_posh_roudi_toml
+    #
+    ######### posh config lib ##########
+    #
+    add_library(iceoryx_posh_config
+        source/log/posh_config_logging.cpp
+        source/gateway/gateway_config.cpp
+        source/gateway/toml_gateway_config_parser.cpp
         source/roudi/roudi_config_toml_file_provider.cpp
+        source/roudi/roudi_config.cpp
+        source/roudi/roudi_cmd_line_parser.cpp
+        source/roudi/roudi_cmd_line_parser_config_file_option.cpp
     )
-    target_link_libraries(iceoryx_posh_roudi_toml PRIVATE cpptoml iceoryx_posh_roudi)
-    set_target_properties(iceoryx_posh_roudi_toml PROPERTIES
+    add_library(${PROJECT_NAMESPACE}::iceoryx_posh_config ALIAS iceoryx_posh_config)
+
+    set_target_properties(iceoryx_posh_config PROPERTIES
         CXX_STANDARD_REQUIRED ON
         CXX_STANDARD ${ICEORYX_CXX_STANDARD}
+        POSITION_INDEPENDENT_CODE ON
+        RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
     )
-    target_compile_options(iceoryx_posh_roudi_toml PRIVATE ${ICEORYX_WARNINGS})
 
+    target_include_directories(iceoryx_posh_config PRIVATE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/generated/iceoryx/include>
+        $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/dependencies/install/include>
+        $<INSTALL_INTERFACE:include/${PREFIX}>
+    )
+
+    target_link_libraries(iceoryx_posh_config PRIVATE
+        iceoryx_utils::iceoryx_utils
+        cpptoml
```

Comment:
currently, the TOML support is not intended to be ASIL D, since the 3rd party lib we are using is heavily unsing the heap, so we have to be careful with the usage

Reply:
This is exactly why this target was created - to isolate it form iceoryx_posh core, which is targeting ASIL D.

Reply:
oh, did you try to build this with -DTOML_CONFIG=OFF? I know of some users who currently use it this way and they want to continue to do so. Additionally, we also need it when we do an ASIL D build